### PR TITLE
Setting Provisioned When VM is running state [#1089]

### PR DESCRIPTION
**Pull request by haagha**
Closes [#1089]

**What this PR does / why we need it**:
Sets VM provisioned state only when client returns its provisioning state as successful

**Special notes for your reviewer**:
If there are any tests needed for this?

**How does this PR make you feel**:
gif

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

File: pkg/resourcemanager/vm/reconcile.go [hunk 1]
```diff
@@ -52,9 +52,14 @@ func (g *AzureVirtualMachineClient) Ensure(ctx context.Context, obj runtime.Obje
 	// to overcome the issue with the lack of idempotence of the Create call
 	item, err := g.GetVirtualMachine(ctx, resourceGroup, resourceName)
 	if err == nil {
-		instance.Status.Provisioned = true
-		instance.Status.Provisioning = false
-		instance.Status.Message = resourcemanager.SuccessMsg
+		if *item.ProvisioningState == "Succeeded" {
```
Comment (huangpf) [inline pkg/resourcemanager/vm/reconcile.go:55]: Is there an enum or const string you can use for the ```"Succeeded"``` string comparison?

Comment (haagha) [reply pkg/resourcemanager/vm/reconcile.go:55]: yes i can fix that

File: pkg/resourcemanager/vm/reconcile.go [hunk 2]
```diff
@@ -52,9 +52,14 @@ func (g *AzureVirtualMachineClient) Ensure(ctx context.Context, obj runtime.Obje
...
+			instance.Status.Provisioned = true
+			instance.Status.Provisioning = false
+			instance.Status.Message = resourcemanager.SuccessMsg
+		} else {
+			instance.Status.Provisioned = false
+			instance.Status.Provisioning = true
+		}
```
Comment (huangpf) [inline pkg/resourcemanager/vm/reconcile.go:62]: @haagha -- I think the following lines should be included in the truly provisioned case?
```
		instance.Status.ResourceId = *item.ID
		return true, nil
```

And the other case should not return?

Comment (haagha) [reply pkg/resourcemanager/vm/reconcile.go:66]: populated this and more status fields for this

**Comment by haagha** [comment]
@frodopwns The pipeline has been running for a couple for hours. Do you know what is average time it takes?

**Comment by haagha** [comment]
rerun CI

**Pull request closed without merging**